packages/vuetify/src/components/VSlideGroup/VSlideGroup.ts

```diff
@@ -287,7 +291,20 @@ export const BaseSlideGroup = mixins<options &
       content.style.setProperty('willChange', 'transform')
     },
     onTouchMove (e: TouchEvent) {
-      this.scrollOffset = this.startX - e.touchmoveX
+      if (!this.isSwiping) {
+        // only calculate disableSwipeHorizontal during the first onTouchMove invoke
+        // in order to ensure disableSwipeHorizontal value is consistent between onTouchStart and onTouchEnd
+        const diffX = e.touchmoveX - e.touchstartX
+        const diffY = e.touchmoveY - e.touchstartY
+        this.isSwipingHorizontal = Math.abs(diffX) > Math.abs(diffY)
+        this.isSwiping = true
+      }
+      if (this.isSwipingHorizontal) {
+        // sliding horizontally
+        this.scrollOffset = this.startX - e.touchmoveX
+        // temporarily disable window vertical scrolling
+        document.documentElement.style.overflowY = 'hidden'
```

Comment:
When swiping horizontally, I tried to achieve disabling vertical scrolling or touch swipe via css by setting `touch-action` to `pan-x`. However, `touch-action` only takes effect from next touch swipe and changing `touch-action` to `pan-x` will not restrict current touch move to "horizontal only". So I use javascript way that disables whole page's vertical scroll if swiping horizontally. (And remove this style in `touchEnd`). Please advice if there is a better way to disable vertical scroll passthrough once horizontal swipe intent is detected